From e95c4ce1b7c925debf4c9b7c794f3b6765edef88 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Thu, 24 Jan 2013 00:48:39 +0100
Subject: BUG/MINOR: cli: show sess should always validate s->listener

Currently s->listener is set for all sessions, but this may not remain
the case forever so we already check s->listener for validity. On check
was missed.

Reported-by: Dinko Korunic <dkorunic@reflected.net>
---
 src/dumpstats.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/dumpstats.c b/src/dumpstats.c
index 24de6d0..94e94f2 100644
--- a/src/dumpstats.c
+++ b/src/dumpstats.c
@@ -3538,7 +3538,7 @@ static int stats_dump_full_sess_to_buffer(struct stream_interface *si, struct se
 			     tm.tm_mday, monthname[tm.tm_mon], tm.tm_year+1900,
 			     tm.tm_hour, tm.tm_min, tm.tm_sec, (int)(sess->logs.accept_date.tv_usec),
 			     sess->uniq_id,
-			     sess->listener->proto->name);
+			     sess->listener && sess->listener->proto->name ? sess->listener->proto->name : "?");
 
 		switch (addr_to_str(&sess->si[0].conn->addr.from, pn, sizeof(pn))) {
 		case AF_INET:
-- 
1.7.1

